server/src/daemon/item.rs

```diff
@@ -0,0 +1,161 @@
+// org.freedesktop.Secret.Item
+
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+};
+
+use oo7::{
+    dbus::api::SecretInner,
+    portal::{self, api::AttributeValue, Keyring},
+};
+use tokio::sync::RwLock;
+use zbus::{
+    object_server::SignalContext,
+    zvariant::{ObjectPath, OwnedObjectPath},
+    ObjectServer,
+};
+
+use super::{
+    collection::Collection, error::ServiceError, prompt::Prompt, service_manager::ServiceManager,
+    Result,
+};
+
+#[derive(Clone, Debug)]
+pub struct Item {
+    inner: Arc<RwLock<portal::Item>>,
+    parameters: Vec<u8>,
+    content_type: String,
+    path: OwnedObjectPath,
+    keyring: Arc<Keyring>,
+    locked: bool,
+    manager: Arc<Mutex<ServiceManager>>,
+}
+
+#[zbus::interface(name = "org.freedesktop.Secret.Item")]
+impl Item {
+    pub async fn delete(
+        &self,
+        #[zbus(object_server)] object_server: &ObjectServer,
+        #[zbus(signal_context)] ctxt: SignalContext<'_>,
+    ) -> Result<Prompt> {
+        let inner = self.inner.read().await;
+        let attributes = inner.attributes();
+        self.keyring
+            .delete(attributes)
+            .await
+            .map_err::<ServiceError, _>(From::from)?;
+        object_server.remove::<Item, _>(self.path()).await?;
+        Collection::item_deleted(&ctxt, self.path()).await?;
+        Ok(Prompt::default())
+    }
+
+    #[zbus(name = "GetSecret")]
+    pub async fn secret(&self, session: ObjectPath<'_>) -> Result<SecretInner> {
+        let inner = self.inner.read().await;
+        let secret = inner.secret();
+        let parameters = self.parameters();
+        let content_type = self.content_type();
+        match self.manager.lock().unwrap().session(session.clone()) {
```

Comment:
instead of clonning, make the session method take a ref, you only need it for comparison no?